x-pack/docs/en/rest-api/logstash/put-pipeline.asciidoc

```diff
@@ -19,25 +24,59 @@ This API creates or updates Pipelines used for Logstash Central Management.
 [[logstash-api-put-pipeline-desc]]
 ==== {api-description-title}
 
-Creates a Logstash Pipeline. If the specified Pipeline exists, the pipeline is
+Creates a {ls} pipeline. If the specified pipeline exists, the pipeline is
 replaced.
 
 [[logstash-api-put-pipeline-params]]
 ==== {api-path-parms-title}
 
 `<pipeline_id>`::
-  (Required, string) Identifier for the Pipeline.
+  (Required, string) Identifier for the pipeline.
+
+[[logstash-api-put-pipeline-request-body]]
+==== {api-request-body-title}
+
+`description`::
+(Optional, string)
+Description of the pipeline. This description is not used by {es} or {ls}.
+
+`last_modified`::
+(Required, string)
+Date the pipeline was last updated. Must be in the `yyyy-MM-dd'T'HH:mm:ss.SSSZZ`
+<<mapping-date-format,`strict_date_time`>> format.
+
+`pipeline`::
+(Required, string)
+Configuration for the pipeline. For supported syntax, see the
+{logstash-ref}/configuration-file-structure.html[{ls} configuration
+documentation].
+
+`pipeline_metadata`::
+(Required, object)
+Optional metadata about the pipeline. May have any contents. This metadata is
+not generated or used by {es} or {ls}.
+
+`pipeline_settings`::
+(Required, object)
+Settings for the pipeline. Supports only flat keys in dot notation. For
+supported settings, see the {logstash-ref}/logstash-settings-file.html[{ls}
+settings documentation].
+
+`username`::
+(Required, string)
+User who last updated the pipeline. This username is not generated or used by
+{es} or {ls}.
```

Comment:
> This username is not generated or used by {es} or {ls}.
@jaymode Can you confirm this is the case? Thanks!

Reply:
As far as I know, this is the case. @kaisecheng do you know of any uses of the username field?

Reply:
@jaymode the `username` display in the pipeline table of central pipeline management, so it is used by Logstash
https://github.com/elastic/kibana/blob/master/x-pack/plugins/logstash/public/application/components/pipeline_list/pipelines_table.js#L59

Reply:
Thanks for confirming @kaisecheng. I removed the related sentence.